# СОБИРАЕМ ИТОГОВЫЙ ДАТАСЕТ

---



In [1]:
import pandas as pd
pd.set_option('display.max_columns', None)

Первый парсер нацеленный на сбор иноформации по API Wildberries - 'Елки_from_0_to_10000000.xlsx' дал нам основу для датасета и для поиска дополнительной информации. Второй парсер отрабытывал ссылки на товар (последний столбец таблицы 'Елки_from_0_to_10000000.xlsx') и дополнительно получал данные о количестве покупок и поле описание, в котором содержатся важные характеристики товара. Отработку страниц вторым парсером делали на 3-х компьютерах поэтому получили 3 файла. Теперь наша задача соединить их в один датасет и выделить значимые характеристики для дальнейшего анализа и построения моделей. 

## ЗАГРУЖАЕМ РЕЗУЛЬТАТЫ РАБОТЫ ПАРСЕРОВ

---



In [2]:
df=pd.read_excel('Елки_from_0_to_10000000.xlsx')

In [3]:
display(df.head(5))
display(df.tail(5))

Unnamed: 0                                       Наименование         id  \
0           0                   Елка Искусственная Сосна с инеем   17371820   
1           1  Ёлка искусственная новогодняя 210 180 150 120 ...   41832891   
2           2                   Елка Искусственная Сосна с инеем   17371819   
3           3                        Ель "Кавказская Люкс" 2,2 м   45732805   
4           4  Ёлка искусственная новогодняя 210 180 150 120 ...  100738560   

   Скидка   Цена  Цена со скидкой                  Бренд  id бренда  \
0      73  11281             3045  Елка Искусственная NG     593476   
1      73   9750             2632                Елки РФ     515496   
2      73   9480             2559  Елка Искусственная NG     593476   
3      15  13552            11519             Новая ёлка      70008   
4      63   8990             3326                Елки РФ     515496   

   feedbacks  rating                                             Ссылка  
0      10969       5  https://www.wildberries.ru/catalog/17371820/de...  
1        931       5  https://www.wildberries.ru/catalog/41832891/de...  
2      10969       5  https://www.wildberries.ru/catalog/17371819/de...  
3        427       5  https://www.wildberries.ru/catalog/45732805/de...  
4        931       5  https://www.wildberries.ru/catalog/100738560/d...

Unnamed: 0                                       Наименование  \
9203        9203  Елка искусственная новогодняя новый год нового...   
9204        9204  Новогодний декор Самая компактная елка украшен...   
9205        9205                                               Елка   
9206        9206                                 Подставка для елки   
9207        9207                                               Елка   

             id  Скидка  Цена  Цена со скидкой        Бренд  id бренда  \
9203  126741447      45  8000             4400   Neon Stars     115951   
9204    4747899      35  2124             1380     BONDIBON       9584   
9205  135046213      61  6500             2535    ART_MAGIC    1346484   
9206   98989539      74  2000              520  Grishingrup   11089253   
9207  135148173      60  6500             2600    ART_MAGIC    1346484   

      feedbacks  rating                                             Ссылка  
9203        138       5  https://www.wildberries.ru/catalog/126741447/d...  
9204         62       5  https://www.wildberries.ru/catalog/4747899/det...  
9205          1       5  https://www.wildberries.ru/catalog/135046213/d...  
9206          1       1  https://www.wildberries.ru/catalog/98989539/de...  
9207          1       5  https://www.wildberries.ru/catalog/135148173/d...

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9208 entries, 0 to 9207
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Unnamed: 0       9208 non-null   int64 
 1   Наименование     9208 non-null   object
 2   id               9208 non-null   int64 
 3   Скидка           9208 non-null   int64 
 4   Цена             9208 non-null   int64 
 5   Цена со скидкой  9208 non-null   int64 
 6   Бренд            9019 non-null   object
 7   id бренда        9208 non-null   int64 
 8   feedbacks        9208 non-null   int64 
 9   rating           9208 non-null   int64 
 10  Ссылка           9208 non-null   object
dtypes: int64(8), object(3)
memory usage: 791.4+ KB


Мы получили 9208 строчек и 10 столбцов, пропуски есть только в столбце бренд, но это не значит что все остальные данные заполнены корректно, но займемся этим потом. Загрузим дополнительную информацию по результатам отработки 2-го парсера

In [5]:
df1=pd.read_excel('ind_from_0_to_5000.xlsx')
df2=pd.read_excel('ind_from_5001_to_7000.xlsx')
df3=pd.read_excel('ind_from_7001_to_10000.xlsx')

In [6]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5001 entries, 0 to 5000
Data columns (total 2 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Количество раз купили  2994 non-null   object
 1   Описание               5001 non-null   object
dtypes: object(2)
memory usage: 78.3+ KB


In [8]:
df1.head(5)

Количество раз купили                                           Описание
0  Купили более 16 500 раз  ['Дополнительная информация\nКонструкция елки\...
1   Купили более 4 900 раз  ['Дополнительная информация\nДлина упаковки\n7...
2  Купили более 21 600 раз  ['Дополнительная информация\nКонструкция елки\...
3   Купили более 2 600 раз  ['Дополнительная информация\nВысота упаковки\n...
4   Купили более 2 100 раз  ['Дополнительная информация\nДлина упаковки\n7...

Мы видим что в результате отработки парсера мы получили 2 оч важных столбца 1) сколько раз купили товар - это может безусловно судить о его популярности и 2) дополнительное описание где через \n содержится черезвычайно полезная информация в очень неодобном формате. Но сначала склеим единый датасет. Для начала склеим результаты отработки 2 го парсера между собой по вертикале (добавляя новые данные снизу)

### Склеиваем 1, 2, 3,  и основной датафрейм

---

In [40]:
df_itog=pd.concat([df1[['Количество раз купили','Описание']], df2[['Количество раз купили','Описание']], df3[['Количество раз купили','Описание']]], ignore_index=True)

In [41]:
df_itog=df_itog[['Количество раз купили','Описание']][0:9208]

In [42]:
df_itog=pd.concat([df, df_itog[['Количество раз купили','Описание']]],axis=1) # так как 2-й парсер отрабытывал по строкам используем самы простой вид объединения - приставление сбоку

In [43]:
df_itog.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9208 entries, 0 to 9207
Data columns (total 13 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   Unnamed: 0             9208 non-null   int64 
 1   Наименование           9208 non-null   object
 2   id                     9208 non-null   int64 
 3   Скидка                 9208 non-null   int64 
 4   Цена                   9208 non-null   int64 
 5   Цена со скидкой        9208 non-null   int64 
 6   Бренд                  9019 non-null   object
 7   id бренда              9208 non-null   int64 
 8   feedbacks              9208 non-null   int64 
 9   rating                 9208 non-null   int64 
 10  Ссылка                 9208 non-null   object
 11  Количество раз купили  4532 non-null   object
 12  Описание               9208 non-null   object
dtypes: int64(8), object(5)
memory usage: 935.3+ KB


In [44]:
display(df_itog.head(5))
display(df_itog.tail(5))

Unnamed: 0                                       Наименование         id  \
0           0                   Елка Искусственная Сосна с инеем   17371820   
1           1  Ёлка искусственная новогодняя 210 180 150 120 ...   41832891   
2           2                   Елка Искусственная Сосна с инеем   17371819   
3           3                        Ель "Кавказская Люкс" 2,2 м   45732805   
4           4  Ёлка искусственная новогодняя 210 180 150 120 ...  100738560   

   Скидка   Цена  Цена со скидкой                  Бренд  id бренда  \
0      73  11281             3045  Елка Искусственная NG     593476   
1      73   9750             2632                Елки РФ     515496   
2      73   9480             2559  Елка Искусственная NG     593476   
3      15  13552            11519             Новая ёлка      70008   
4      63   8990             3326                Елки РФ     515496   

   feedbacks  rating                                             Ссылка  \
0      10969       5  https://www.wildberries.ru/catalog/17371820/de...   
1        931       5  https://www.wildberries.ru/catalog/41832891/de...   
2      10969       5  https://www.wildberries.ru/catalog/17371819/de...   
3        427       5  https://www.wildberries.ru/catalog/45732805/de...   
4        931       5  https://www.wildberries.ru/catalog/100738560/d...   

     Количество раз купили                                           Описание  
0  Купили более 16 500 раз  ['Дополнительная информация\nКонструкция елки\...  
1   Купили более 4 900 раз  ['Дополнительная информация\nДлина упаковки\n7...  
2  Купили более 21 600 раз  ['Дополнительная информация\nКонструкция елки\...  
3   Купили более 2 600 раз  ['Дополнительная информация\nВысота упаковки\n...  
4   Купили более 2 100 раз  ['Дополнительная информация\nДлина упаковки\n7...

Unnamed: 0                                       Наименование  \
9203        9203  Елка искусственная новогодняя новый год нового...   
9204        9204  Новогодний декор Самая компактная елка украшен...   
9205        9205                                               Елка   
9206        9206                                 Подставка для елки   
9207        9207                                               Елка   

             id  Скидка  Цена  Цена со скидкой        Бренд  id бренда  \
9203  126741447      45  8000             4400   Neon Stars     115951   
9204    4747899      35  2124             1380     BONDIBON       9584   
9205  135046213      61  6500             2535    ART_MAGIC    1346484   
9206   98989539      74  2000              520  Grishingrup   11089253   
9207  135148173      60  6500             2600    ART_MAGIC    1346484   

      feedbacks  rating                                             Ссылка  \
9203        138       5  https://www.wildberries.ru/catalog/126741447/d...   
9204         62       5  https://www.wildberries.ru/catalog/4747899/det...   
9205          1       5  https://www.wildberries.ru/catalog/135046213/d...   
9206          1       1  https://www.wildberries.ru/catalog/98989539/de...   
9207          1       5  https://www.wildberries.ru/catalog/135148173/d...   

     Количество раз купили                                           Описание  
9203   Купили более 80 раз  ['Дополнительная информация\nШирина упаковки\n...  
9204  Купили более 900 раз  ['Дополнительная информация\nСтрана производст...  
9205    Купили более 5 раз  ['Дополнительная информация\nВысота упаковки\n...  
9206   Купили более 10 раз  ['Дополнительная информация\nВес с упаковкой (...  
9207    Купили менее 5 раз  ['Дополнительная информация\nВысота упаковки\n...

Все подгрузилось корректно. Обращаем внимание на то что данные о покупках есть только по 4532 строкам

### ЗАПИШЕМ ПОЛУЧИВШИЕСЯ ДАННЫЕ В ФАЙЛ

In [ ]:
filename=f'final_uncleaned_data_ind_from_{0}_to_{9207}'
writer = pd.ExcelWriter(f'{filename}.xlsx')
df_itog.to_excel(writer, 'data', index=False)
writer.save()

### ЧИСТИМ ДАННЫЕ И ВЫДЕЛЯЕМ ДОПОЛНИТЕЛЬНЫЕ ПРИЗНАКИ

В ячейке "Количество раз купили" вычленяем значение 

---





In [45]:
def number_of_times_bought(st):
    try:
        if int(st.split(' ')[-3])>0:
            num=int(st.split(' ')[-3])*1000+int(st.split(' ')[-2])
            return num
    except:
        try:
            if int(st.split(' ')[-2])>0:
                num=int(st.split(' ')[-2])
                return num
        except:
            if st=='Error':
                return 'Error'

In [46]:
df_itog['Количество раз купили']=df_itog['Количество раз купили'].apply(number_of_times_bought)

In [47]:
df_itog['Количество раз купили'].unique()

array([16500, 4900, 21600, 2600, 2100, 1200, 2800, 800, 3400, 700, 500,
       200, 1000, 5800, 100, 10600, 900, 8100, 5, 10, 20, 90, 2000, 5500,
       400, 1400, 30, 300, 50, 60, 600, 4400, 40, 80, 70, 4100, None,
       'Error', 3700, 5300, 9500, 7800, 4600, 12700, 3000, 2200, 1600,
       4200, 1500, 4300, 2900, 3300, 1100, 6600, 2500, 1300, 3500, 1700,
       1900, 3200, 2300, 2700, 8900, 4000, 3600, 1800, 2400, 5000],
      dtype=object)

Посчитаем предварительно количество категорий по краткому описанию из 
ячейки 'Дополнительная информация'


---



In [48]:
words={}
n_error=0
for i in range(8001):
    try:
        for j in df_itog['Описание'][i][2:-2].split('Дополнительная информация')[1].split('\\n')[1::2]:
            
            if words.get(j, 0)==0:
                words[j]=1
            else:
                words[j]=words.get(j, 0)+1
    except:
        n_error+=1

Количество ошибок полученных при парсинге данных со страниц товара

---



In [49]:
n_error

422

In [50]:
words

{'Конструкция елки': 1809,
 'Подсветка': 557,
 'Страна производства': 2038,
 'Высота упаковки': 5921,
 'Длина упаковки': 5986,
 'Особенности елки': 1108,
 'Тип елки': 961,
 'Высота елки': 2724,
 'Ширина упаковки': 5808,
 'Вес с упаковкой (кг)': 3843,
 'Назначение подарка': 1237,
 'Материал елки': 1559,
 'Повод': 1281,
 'Количество ветвей': 1394,
 'Комплектация': 388,
 'Ограничение для дома': 9,
 'Материал подставки': 45,
 'Высота предмета': 72,
 'Вес товара без упаковки (г)': 107,
 'Ширина предмета': 43,
 'Ограничение без ограничений': 4,
 'Максимальная нагрузка': 10,
 'Глубина упаковки': 9,
 'Вес товара с упаковкой (г)': 51,
 'Ограничение нет': 4,
 'Ограничение 3+': 1,
 'Ограничение нет ограничений / без ограничений': 2,
 'Ограничение Нет ограничений/Подходит всем': 1,
 'Ограничение Елка новогодняя искусственная в квартиру': 1,
 'Ограничение Нет ограничений к ножу': 1,
 'Ограничение Без огранчений': 1,
 'Ограничение не хрупкое': 1}

Посчитаем количество фитчей в дополнительном описании

---



In [51]:
words_dop={x: 0 for x in list(words.keys())}


In [52]:
for i in range(len(df_itog['Описание'])):
    try:
        for word in df_itog['Описание'][i][2:-2].split('Дополнительная информация')[2].split('\\n'):
            k=0
            for fitch in list(words_dop.keys()):

                if fitch in word:
                    k+=1
                    words_dop[fitch]=words_dop.get(fitch, 0)+1
                 
            if k==0:
                words_dop[word]=words_dop.get(word, 0)+1
                               
    except:
        continue

In [53]:
words_dop

{'Конструкция елки': 4067,
 'Подсветка': 1259,
 'Страна производства': 3415,
 'Высота упаковки': 7226,
 'Длина упаковки': 7354,
 'Особенности елки': 2153,
 'Тип елки': 1941,
 'Высота елки': 4027,
 'Ширина упаковки': 7154,
 'Вес с упаковкой (кг)': 4787,
 'Назначение подарка': 2682,
 'Материал елки': 3773,
 'Повод': 2445,
 'Количество ветвей': 2410,
 'Комплектация': 1147,
 'Ограничение для дома': 0,
 'Материал подставки': 125,
 'Высота предмета': 145,
 'Вес товара без упаковки (г)': 114,
 'Ширина предмета': 65,
 'Ограничение без ограничений': 0,
 'Максимальная нагрузка': 45,
 'Глубина упаковки': 9,
 'Вес товара с упаковкой (г)': 118,
 'Ограничение нет': 0,
 'Ограничение 3+': 0,
 'Ограничение нет ограничений / без ограничений': 0,
 'Ограничение Нет ограничений/Подходит всем': 0,
 'Ограничение Елка новогодняя искусственная в квартиру': 0,
 'Ограничение Нет ограничений к ножу': 0,
 'Ограничение Без огранчений': 0,
 'Ограничение не хрупкое': 0,
 '': 65249}

In [54]:
set(words_dop)-set(words)

{''}

Критерием для итогово списка признаков, которые мы оставим, будет не менее 1000

---





In [55]:
feature_list=[x for x in list(words_dop.keys()) if words_dop[x]>1000 and words_dop[x]<10000]

In [56]:
feature_list

['Конструкция елки',
 'Подсветка',
 'Страна производства',
 'Высота упаковки',
 'Длина упаковки',
 'Особенности елки',
 'Тип елки',
 'Высота елки',
 'Ширина упаковки',
 'Вес с упаковкой (кг)',
 'Назначение подарка',
 'Материал елки',
 'Повод',
 'Количество ветвей',
 'Комплектация']

In [57]:
len(feature_list)

15

In [58]:
def add_dict(description):
    feature_list=['Конструкция елки',
 'Подсветка',
 'Страна производства',
 'Высота упаковки',
 'Длина упаковки',
 'Особенности елки',
 'Тип елки',
 'Высота елки',
 'Ширина упаковки',
 'Вес с упаковкой (кг)',
 'Назначение подарка',
 'Материал елки',
 'Повод',
 'Количество ветвей',
 'Комплектация']
    
    feature_list_cell=[None]*15

    try:
        #длина фитчей в предварительном описании
        pred_len=len(description[2:-2].split('Дополнительная информация')[1].split('\\n')[2::2]) 
        
        #длина фитчей в полном описаниие        
        poln_len=len(description[2:-2].split('Дополнительная информация')[2].split('\\n')[1:])

        #добавляем в список все фитчи в предварительном описании
        for i in range(len(feature_list)):
            for j in range(pred_len):

                if feature_list[i]==description[2:-2].split('Дополнительная информация')[1].split('\\n')[1::2][j]:
                    feature_list_cell[i]=(description[2:-2].split('Дополнительная информация')[1].split('\\n')[2::2][j])

        #проверяем остались ли фитчи в дополнительном описании и добавляем их
        if pred_len < poln_len:             
            for i in range(len(feature_list)):
                hm=pred_len-poln_len
                for j in range(hm, 0, 1):
                    st=feature_list[i]

                    if st in description[2:-2].split('Дополнительная информация')[2].split('\\n')[1:][j]:
                        feature_list_cell[i]=description[2:-2].split('Дополнительная информация')[2].split('\\n')[1:][j].split(st+' ')[1]
        return {x:y for x,y in zip(feature_list, feature_list_cell)}  
        
    except:
        return {x:y for x,y in zip(feature_list, ['Error']*15)}


In [59]:
df_itog['Список с значениями']=df_itog['Описание'].apply(add_dict)

#### Удаляем ячейки со значениями Error

In [60]:
df_itog=df_itog[(df_itog['Количество раз купили']!='Error')&(df_itog['Описание']!='Error')]

In [61]:
df_itog

Unnamed: 0                                       Наименование  \
0              0                   Елка Искусственная Сосна с инеем   
1              1  Ёлка искусственная новогодняя 210 180 150 120 ...   
2              2                   Елка Искусственная Сосна с инеем   
3              3                        Ель "Кавказская Люкс" 2,2 м   
4              4  Ёлка искусственная новогодняя 210 180 150 120 ...   
...          ...                                                ...   
9203        9203  Елка искусственная новогодняя новый год нового...   
9204        9204  Новогодний декор Самая компактная елка украшен...   
9205        9205                                               Елка   
9206        9206                                 Подставка для елки   
9207        9207                                               Елка   

             id  Скидка   Цена  Цена со скидкой                  Бренд  \
0      17371820      73  11281             3045  Елка Искусственная NG   
1      41832891      73   9750             2632                Елки РФ   
2      17371819      73   9480             2559  Елка Искусственная NG   
3      45732805      15  13552            11519             Новая ёлка   
4     100738560      63   8990             3326                Елки РФ   
...         ...     ...    ...              ...                    ...   
9203  126741447      45   8000             4400             Neon Stars   
9204    4747899      35   2124             1380               BONDIBON   
9205  135046213      61   6500             2535              ART_MAGIC   
9206   98989539      74   2000              520            Grishingrup   
9207  135148173      60   6500             2600              ART_MAGIC   

      id бренда  feedbacks  rating  \
0        593476      10969       5   
1        515496        931       5   
2        593476      10969       5   
3         70008        427       5   
4        515496        931       5   
...         ...        ...     ...   
9203     115951        138       5   
9204       9584         62       5   
9205    1346484          1       5   
9206   11089253          1       1   
9207    1346484          1       5   

                                                 Ссылка Количество раз купили  \
0     https://www.wildberries.ru/catalog/17371820/de...                 16500   
1     https://www.wildberries.ru/catalog/41832891/de...                  4900   
2     https://www.wildberries.ru/catalog/17371819/de...                 21600   
3     https://www.wildberries.ru/catalog/45732805/de...                  2600   
4     https://www.wildberries.ru/catalog/100738560/d...                  2100   
...                                                 ...                   ...   
9203  https://www.wildberries.ru/catalog/126741447/d...                    80   
9204  https://www.wildberries.ru/catalog/4747899/det...                   900   
9205  https://www.wildberries.ru/catalog/135046213/d...                     5   
9206  https://www.wildberries.ru/catalog/98989539/de...                    10   
9207  https://www.wildberries.ru/catalog/135148173/d...                     5   

                                               Описание  \
0     ['Дополнительная информация\nКонструкция елки\...   
1     ['Дополнительная информация\nДлина упаковки\n7...   
2     ['Дополнительная информация\nКонструкция елки\...   
3     ['Дополнительная информация\nВысота упаковки\n...   
4     ['Дополнительная информация\nДлина упаковки\n7...   
...                                                 ...   
9203  ['Дополнительная информация\nШирина упаковки\n...   
9204  ['Дополнительная информация\nСтрана производст...   
9205  ['Дополнительная информация\nВысота упаковки\n...   
9206  ['Дополнительная информация\nВес с упаковкой (...   
9207  ['Дополнительная информация\nВысота упаковки\n...   

                                    Список с значениями  
0     {'Конструкция елки': 'сборная не литая; не нас...  
1     {'К

Добавляем значения полученне в словаре в отдельных 15 колонок


---



In [62]:
new_df = df_itog['Список с значениями'].apply(pd.Series)

In [63]:
final_df = pd.concat([df_itog, new_df],axis=1)

Удаляем столбцы Описание, Список с значениями и 'Unnamed: 0'

---



In [64]:
final_df.drop(columns = ['Список с значениями', 'Описание'],axis = 1, inplace=True)

In [65]:
final_df=final_df.reset_index(drop=True)

In [66]:
final_df.drop(columns = ['Unnamed: 0'],axis = 1, inplace=True)

In [68]:
display(final_df)

Наименование         id  Скидка  \
0                      Елка Искусственная Сосна с инеем   17371820      73   
1     Ёлка искусственная новогодняя 210 180 150 120 ...   41832891      73   
2                      Елка Искусственная Сосна с инеем   17371819      73   
3                           Ель "Кавказская Люкс" 2,2 м   45732805      15   
4     Ёлка искусственная новогодняя 210 180 150 120 ...  100738560      63   
...                                                 ...        ...     ...   
8782  Елка искусственная новогодняя новый год нового...  126741447      45   
8783  Новогодний декор Самая компактная елка украшен...    4747899      35   
8784                                               Елка  135046213      61   
8785                                 Подставка для елки   98989539      74   
8786                                               Елка  135148173      60   

       Цена  Цена со скидкой                  Бренд  id бренда  feedbacks  \
0     11281             3045  Елка Искусственная NG     593476      10969   
1      9750             2632                Елки РФ     515496        931   
2      9480             2559  Елка Искусственная NG     593476      10969   
3     13552            11519             Новая ёлка      70008        427   
4      8990             3326                Елки РФ     515496        931   
...     ...              ...                    ...        ...        ...   
8782   8000             4400             Neon Stars     115951        138   
8783   2124             1380               BONDIBON       9584         62   
8784   6500             2535              ART_MAGIC    1346484          1   
8785   2000              520            Grishingrup   11089253          1   
8786   6500             2600              ART_MAGIC    1346484          1   

      rating                                             Ссылка  \
0          5  https://www.wildberries.ru/catalog/17371820/de...   
1          5  https://www.wildberries.ru/catalog/41832891/de...   
2          5  https://www.wildberries.ru/catalog/17371819/de...   
3          5  https://www.wildberries.ru/catalog/45732805/de...   
4          5  https://www.wildberries.ru/catalog/100738560/d...   
...      ...                                                ...   
8782       5  https://www.wildberries.ru/catalog/126741447/d...   
8783       5  https://www.wildberries.ru/catalog/4747899/det...   
8784       5  https://www.wildberries.ru/catalog/135046213/d...   
8785       1  https://www.wildberries.ru/catalog/98989539/de...   
8786       5  https://www.wildberries.ru/catalog/135148173/d...   

     Количество раз купили                 Конструкция елки  \
0                    16500  сборная не литая; не настольная   
1                     4900                             None   
2                    21600  сборная не литая; не настольная   
3                     2600                        ствольная   
4                     2100                             None   
...                    ...                              ...   
8782                    80                        ствольная   
8783                   900                             None   
8784                     5                             None   
8785                    10                             None   
8786                     5                             None   

              Подсветка Страна производства Высота упаковки Длина упаковки  \
0     украшения на елку               Китай           19 см      79 см', '   
1              Гирлянда                None            None          70 см   
2     украшения на елку               Китай           17 см      75 см', '   
3                  None                None           19 см         119 см   
4              Гирлянда                None       20 см', '          70 см   
...                 ...                 ...             ...            ...   
8782               None               Китай            None           Non

ЗАПИСЫВАЕМ ИТОГ В ПРОМЕЖУТОЧНЫЙ РЕЗУЛЬТАТ

---



In [ ]:
filename=f'final_data_for_cleaning_and_working_with_gaps'
writer = pd.ExcelWriter(f'{filename}.xlsx')
final_df.to_excel(writer, 'data', index=False)
writer.save()

# ДАЛЕЕ НАС ЖДУТ КРОПОТЛИВАЯ ОЧИСТКА ДАННЫХ

---

